### PR TITLE
fix(api): refactor block queries to avoid lookups

### LINKED ISSUE #1105: API endpoints which request blocks will not return conflicting outputs
### Bug description

API endpoints like `core/v2/block`, if not requesting the raw block, will return blocks without the outputs if the transaction is conflicting.

### Chronicle version

- Chronicle version (version number, commit, or branch): 1.0.0-beta.30

### PR DESCRIPTION
**Pull request by Alexandcoats**
## Linked Issues

* Closes #1105 

## Notes to Reviewer

As a reviewer, please pay particular attention to the following areas when reviewing this PR and tick the above boxes after you have completed the steps.

#### API Changes
* [x] Test the API endpoints which were added/changed.
* [x] Ensure that the API response times scale with database size appropriately.
* [x] Review the API documentation changes and confirm that it matches the actual functionality.
* [x] Check for breaking changes in the API and matching (conventional) commit message prefix.

#### Database Changes
* [x] Review database queries for correctness/conciseness.
* [x] Ensure queries are supported by indexes if needed.
* [x] Check for breaking changes in the data model and matching (conventional) commit message prefix.

### CODE REVIEW

File: src/bin/inx-chronicle/api/stardust/poi/routes.rs [hunk 1]
```diff
@@ -179,7 +179,7 @@ async fn create_proof_for_applied_blocks(
 
     // Fetch the referenced and applied block ids in "White Flag" order, and make sure they contain the block.
     let applied_block_ids = block_collection
-        .get_applied_blocks_in_white_flag_order(referenced_index)
```
Comment (Alex6323) [inline src/bin/inx-chronicle/api/stardust/poi/routes.rs:182]: How is that not breaking PoI for applied transactions? :thinking:

File: src/db/collections/block.rs [hunk 2]
```diff
@@ -132,64 +131,47 @@ pub struct IncludedBlockResult {
 
 #[derive(Deserialize, Debug, Clone)]
 pub struct IncludedBlockMetadataResult {
+    #[serde(rename = "_id")]
```
Comment (Alex6323) [inline src/db/collections/block.rs:134]: What about `IncludedBlockResult`? Should we rename there as well for consistency?

File: src/db/collections/block.rs [hunk 3]
```diff
@@ -132,64 +131,47 @@ pub struct IncludedBlockResult {
...
-            ],
-            None,
-        )
-        .await?
-        .try_next()
-        .await
+        Ok(self
+            .get_block_raw(block_id)
+            .await?
+            .map(|raw| iota_types::block::Block::unpack_unverified(raw).unwrap().into()))
```
Comment (Alex6323) [inline src/db/collections/block.rs:158]: I have a very bad feeling about this line.

Comment (Alexandcoats) [reply src/db/collections/block.rs:158]: How so?

File: src/bin/inx-chronicle/api/stardust/poi/routes.rs (hunk 1 above)
Comment (Alexandcoats) [reply src/bin/inx-chronicle/api/stardust/poi/routes.rs:182]: The problem is the `Included` check? I think I was assuming referenced blocks are never conflicting.

Comment (Alex6323) [reply src/bin/inx-chronicle/api/stardust/poi/routes.rs:182]: Yeah...that's the problem. Milestones can reference conflicting transactions.

File: src/db/collections/block.rs (hunk 3 above)
Comment (Alex6323) [reply src/db/collections/block.rs:159]: Well ... what happens if the binary representation changes in the future? How do we then unpack old Blocks if we do it unverified?

File: src/bin/inx-chronicle/api/stardust/poi/routes.rs (hunk 1 above)
Comment (Alexandcoats) [reply src/bin/inx-chronicle/api/stardust/poi/routes.rs:182]: Makes sense, my mistake. I'll revert this change.

File: src/db/collections/block.rs (hunk 3 above)
Comment (Alexandcoats) [reply src/db/collections/block.rs:159]: If the representation changes, there will have to be a version change in `iota_types`, and most likely there will have to be a solution in that crate for old representations.

Comment (Alex6323) [reply src/db/collections/block.rs:159]: I am unsure here. Arent' the protocol parameters all it needs as context to unpack older versions of blocks?

Comment (Alex6323) [reply src/db/collections/block.rs:159]: right, thx

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**